### PR TITLE
Use discriminator key from inheritance attribute when serializing

### PR DESCRIPTION
**Pull request by bgorven**
This ensures that the key that is written matches the key that was read.

### CODE REVIEW

**Comment by RSuter** [comment]
Thanks for the PR!

File: src/NJsonSchema.CodeGeneration.Tests/InheritanceSerializationTests.cs [hunk 1]
```diff
@@ -259,5 +263,76 @@ public void JsonInheritanceConverter_is_thread_safe()
...
+                        .Where(diagnostic => diagnostic.IsWarningAsError || diagnostic.Severity == Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                        .Select(d => d.Location.GetLineSpan().StartLinePosition + " - " + d.GetMessage())) + "\n" + code);
+                }
+
+                return Assembly.Load(stream.GetBuffer());
+            }
+#endif
+
+#if NET451
+            var provider = new CSharpCodeProvider();
```
Comment (stijnherreman) [inline src/NJsonSchema.CodeGeneration.Tests/InheritanceSerializationTests.cs]: This needs to be fully qualified as `Microsoft.CSharp.CSharpCodeProvider`, or a `using` needs to be added. The code doesn't build as-is.

Comment (RSuter) [reply src/NJsonSchema.CodeGeneration.Tests/InheritanceSerializationTests.cs]: Thanks for finding this problem

Comment (bgorven) [reply src/NJsonSchema.CodeGeneration.Tests/InheritanceSerializationTests.cs]: I'm sorry I missed that.

**Pull request merged**